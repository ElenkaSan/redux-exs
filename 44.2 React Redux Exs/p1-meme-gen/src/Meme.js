import React from "react";
import PropTypes from "prop-types";
import Button from '@mui/material/Button';
import Card from '@mui/material/Card';
import CardMedia from '@mui/material/CardMedia';
import CardActions from '@mui/material/CardActions';
import Grid from '@mui/material/Grid';

function Meme({ deleteMeme, urlLink, topText, bottomText, id }) {
    function handleDelete() {
        deleteMeme(id);
    }

    return (
        <Grid container justifyContent="center" >
            <Card sx={{ maxWidth: 400 }} className="card">
              <span className="top">{topText}</span> 
              <CardMedia
               component="img"
               height="400"
               width="400"
               src={urlLink}
               alt="meme"
              />
             <span className="bottom">{bottomText}</span>
             <CardActions color="primary">
               <Button variant="outlined" color="error" size="large" onClick={handleDelete}>Delete me</Button>
             </CardActions>
            </Card>
        </Grid>
    )
}

Meme.propTypes = {
    topText: PropTypes.string.isRequired,
    urlLink: PropTypes.string.isRequired,
    bottomText: PropTypes.string.isRequired
};

export default Meme;