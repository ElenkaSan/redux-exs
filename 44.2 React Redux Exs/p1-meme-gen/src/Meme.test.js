import React from 'react';
import { render, screen } from '@testing-library/react';
import Meme from './Meme';

let mem;
beforeEach(() => {
  mem = render(
    <Meme
      urlLink="https://media2.giphy.com/media/KPZhv4wNSpaJNfuAzM/200w.webp"
      topText="Haha check"
      bottomText="No way"
    />
  );
})
it('renders', () => {
  expect(mem).not.toBeNull();
});
it('matches snapshot', () => {
  expect(mem.asFragment()).toMatchSnapshot();
});
it('applies url link prop to image src alt', () => {
  const img = screen.getByAltText('meme');
  
  expect(img).toBeInTheDocument();
  expect(img).toHaveAttribute(
    'src',
    'https://media2.giphy.com/media/KPZhv4wNSpaJNfuAzM/200w.webp'
  );
});
