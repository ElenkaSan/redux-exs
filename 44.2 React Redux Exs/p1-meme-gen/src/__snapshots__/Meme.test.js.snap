// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-qpjsw6-MuiPaper-root-MuiCard-root"
    >
      <span
        class="top"
      >
        Haha check
      </span>
      <img
        alt="meme"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="400"
        src="https://media2.giphy.com/media/KPZhv4wNSpaJNfuAzM/200w.webp"
        width="400"
      />
      <span
        class="bottom"
      >
        No way
      </span>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        color="primary"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeLarge MuiButton-outlinedSizeLarge css-eyt2ms-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Delete me
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
